### ISSUE TITLE
Dependabot doesn't recognize updated Docker images.

### ISSUE THREAD

**Issue by oittaa** [dependabot/dependabot-core#3929]
dependabot.yml

```
version: 2
updates:
- package-ecosystem: pip
  directory: "/app"
  schedule:
    interval: daily
  open-pull-requests-limit: 10
- package-ecosystem: "docker"
  directory: "/app"
  schedule:
    interval: "daily"
  open-pull-requests-limit: 10
- package-ecosystem: "github-actions"
  directory: "/"
  schedule:
    interval: "daily"
  open-pull-requests-limit: 10
  
```

https://github.com/oittaa/avif-converter/network/updates/163792800

important part from the log:
```
updater | INFO <job_163792800> Checking if oittaa/imagemagick 7.0.11-14 needs updating
...
updater | INFO <job_163792800> Latest version is 7.0.11-14
updater | INFO <job_163792800> No update needed for oittaa/imagemagick 7.0.11-14
updater | INFO <job_163792800> Finished job processing
```

Even though there's version 7.1.0-0 which should be used: 
https://hub.docker.com/r/oittaa/imagemagick/tags?page=1&ordering=name

**Comment by jurre** [comment]
In order for us to be able to help you better, could you please include a Dockerfile that reproduces the issue?

**Comment by oittaa** [comment]
> In order for us to be able to help you better, could you please include a Dockerfile that reproduces the issue?

https://github.com/oittaa/avif-converter/blob/1660d70ff5572cf0aa4cddb33a5873a6cba8ccf2/app/Dockerfile

**Comment by jurre** [comment]
Thanks! I just had a look and I noticed the tags are specified as `7.0.11-14`, with a `dash` that _seems_ to represent some incrementing version on top of the semver part of the tag. Typically this format is used to specify the part before the dash as a prefix, where at least _one_ of those is expected to be static (e.g `trusty-20170817`).

I'm not sure if there is a way for us to determine what the right thing to is in this scenario (it's clear for this _exact_ example, but I'm talking about in general). If I'd fix our implementation to match the semver part (`7.0.11`), we'd end up never incrementing the numeric version.

I'll need to marinate on this for a bit.

**Comment by oittaa** [comment]
It's not the cleanest versioning scheme for Docker images, but I'm just using the version numbers that the original [ImageMagick](https://github.com/ImageMagick/ImageMagick) project uses. Otherwise it would be a nightmare to track what versions my Docker images are actually using.

**Comment by oittaa** [comment]
> Typically this format is used to specify the part before the dash as a prefix, where at least one of those is expected to be static (e.g trusty-20170817).

That's not always the case. For example Linux packages have a dash part after semver quite regularly.

Firefox in Debian: https://packages.debian.org/search?keywords=firefox&searchon=names&suite=all&section=all
OpenSSL in Debian: https://packages.debian.org/search?keywords=openssl&searchon=names&suite=all&section=all

**Comment by trinitronx** [comment]
I encountered the same issue using _private Artifactory_ Docker Registry while trying to update an image tagged `5.7.36` to new tag: `5.7.37`.

Seems that the `/v2/<image_name_here>/tags/list` Docker Registry API endpoint was only returning `SHA256:*` tags.  I wasn't sure whether dependabot was resolving these tags to their friendly named tag "aliases" (nothing logged as such).  Dependabot logged that it did not need to update the image, although Artifactory's `/artifactory/list/docker/<image_name_here` API endpoint did return all those tags, including the new "friendly named" version: `5.7.37`.

Dependabot logs showing that it did not think the image needed updating:

```
updater | INFO <job_316112552> Checking if mysql 5.7.36 needs updating
  proxy | 2022/03/17 00:57:52 [012] GET <https://**SCRUBBED ${ARTIFACTORY_HOST}**:443/v2/mysql/tags/list>
  proxy | 2022/03/17 00:57:52 [012] * authenticating docker registry request (host: **SCRUBBED ${ARTIFACTORY_HOST}**)
  proxy | 2022/03/17 00:57:56 [012] 200 <https://**SCRUBBED ${ARTIFACTORY_HOST}**:443/v2/mysql/tags/list>
updater | INFO <job_316112552> Latest version is 5.7.36
updater | INFO <job_316112552> No update needed for mysql 5.7.36
updater | INFO <job_316112552> Finished job processing
updater | time="2022-03-17T00:57:56Z" level=info msg="task complete" container_id=job-316112552-updater exit_code=0 job_id=316112552 step=updater
```

**Comment by jeffwidman** [comment]
Related:
* https://github.com/dependabot/dependabot-core/issues/4329
* https://github.com/dependabot/dependabot-core/issues/5538
* https://github.com/dependabot/dependabot-core/pull/5560

**Comment by deivid-rodriguez** [comment]
This will be fixed by #6173!

**Comment by justinmchase** [comment]
> I just had a look and I noticed the tags are specified as 7.0.11-14, with a dash that seems to represent some incrementing version on top of the semver part of the tag.

@jurre I just want to point out that `7.0.11-14` is a valid semantic version and `-14` is actually part of the semantic version known as the pre-release information.

The valid syntax for it is essentially `-(alphanum+)(\.alphanum+)*`.

You can see it in the [semver](https://semver.org/#summary) spec Summary as:

> Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

Real briefly `7.0.11 > 7.0.11-14 > 7.0.10`. Or in English, pre-release versions are _less than_ the same non-prerelease version but more than all lower non-prerelease versions.

Semver pre-release spec [9](https://semver.org/#spec-item-9):

> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. Numeric identifiers MUST NOT include leading zeroes. Pre-release versions have a lower precedence than the associated normal version. A pre-release version indicates that the version is unstable and might not satisfy the intended compatibility requirements as denoted by its associated normal version. Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, 1.0.0-x-y-z.--.
